pkg/backend/diy/backend.go

```diff
@@ -515,8 +515,29 @@ func massageBlobPath(path string) (string, error) {
 		return path, nil
 	}
 
+	// We need to set no_tmp_dir to a value to avoid using the system temp directory.
+	// See also https://github.com/pulumi/pulumi/issues/15352
+	url, err := url.Parse(path)
+	if err != nil {
+		return "", fmt.Errorf("Could not parse the provided URL: %w", err)
```

Comment:
Nit:

```suggestion
		return "", fmt.Errorf("parsing the provided URL: %w", err)
```

See https://go.dev/wiki/CodeReviewComments#error-strings

Reply:
I keep meaning to do a pass over the whole codebase for strings like this

Reply:
And then turn a linter on for it